core/src/main/java/org/mule/api/config/Config.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.config;
+
+/**
+ * A generic component's configuration. For now it's just a marker to enable
+ * a lifecycle mechanism in which configs are initialised and started before the
+ * owning component. This interface should not be used in extensions or applications
```

Comment:
yes... the wording on this comment is wrong. It shouldn't be used in applications but should in extensions. I will remove that sentence.